src/components/Engines/Engines.tsx

```diff
@@ -1,73 +1,51 @@
+import React, { useContext } from 'react';
 import Grid from '@material-ui/core/Grid';
 import ListItemText from '@material-ui/core/ListItemText';
 
-import { VersionPageConsumerProps, DetailContextConsumer } from '../../pages/Version';
+import { DetailContext } from '../../pages/Version';
 import Avatar from '../../muiComponents/Avatar';
 import List from '../../muiComponents/List';
 import npm from '../Install/img/npm.svg';
 
 import { StyledText, EngineListItem } from './styles';
-// @ts-ignore
 import node from './img/node.png';
 
-const ICONS = {
-  'node-JS': <Avatar src={node} />,
-  'NPM-version': <Avatar src={npm} />,
-};
-
-class Engine extends Component {
-  public render(): ReactElement<HTMLElement> {
-    return (
-      <DetailContextConsumer>
-        {context => {
-          return this.renderEngine(context as VersionPageConsumerProps);
-        }}
-      </DetailContextConsumer>
-    );
-  }
-
-  private renderEngine = ({ packageMeta }): ReactElement<HTMLElement> | null => {
-    const { engines } = packageMeta.latest;
-    if (!engines) {
-      return null;
-    }
+const Engine: React.FC = () => {
+  const { packageMeta } = useContext(DetailContext);
 
-    const engineDict = {
-      'node-JS': engines.node,
-      'NPM-version': engines.npm,
-    };
+  const engines = packageMeta && packageMeta.latest && packageMeta.latest.engines;
 
-    const accumulator: React.ReactNode[] = [];
-    const items = Object.keys(engineDict).reduce((markup, text, key) => {
-      const heading = engineDict[text];
-      if (heading) {
-        markup.push(
-          <Grid item={true} key={key} xs={6}>
-            {this.renderListItems(heading, text)}
-          </Grid>
-        );
-      }
-      return markup;
-    }, accumulator);
-
-    if (items.length < 1) {
-      return null;
-    }
-
-    return <Grid container={true}>{items}</Grid>;
-  };
+  if (!engines || (!engines.node && !engines.npm)) {
+    return null;
+  }
 
-  private renderListItems = (heading: string, text: string) => {
-    return (
-      <List subheader={<StyledText variant={'subtitle1'}>{text.split('-').join(' ')}</StyledText>}>
-        <EngineListItem button={true}>
-          {ICONS[text]}
-          <ListItemText primary={heading} />
-        </EngineListItem>
-      </List>
-    );
-  };
-}
+  /* eslint-disable react/jsx-max-depth */
+  return (
+    <Grid container={true}>
+      {engines.node && (
+        <Grid item={true} xs={6}>
+          <List subheader={<StyledText variant={'subtitle1'}>{'node JS'}</StyledText>}>
+            <EngineListItem button={true}>
+              <Avatar src={node} />
+              <ListItemText primary={engines.node} />
+            </EngineListItem>
+          </List>
+        </Grid>
+      )}
+
+      {engines.npm && (
+        <Grid item={true} xs={6}>
+          <List subheader={<StyledText variant={'subtitle1'}>{'NPM version'}</StyledText>}>
+            <EngineListItem button={true}>
```

Comment:
Same here, You can remove button={true}

Reply:
This should be enforced by `eslint`, I'd suggest let it go and add the plugin or rule and change all places at a time.

Reply:
I think @ayusharma is referring to the fact that the `EngineListItem` styled component doesn't accept a `button` prop.

I left it because removing it breaks the snapshot tests, which I consider bad practice when doing a refactor like this.

Reply:
Hi all! I just checked and the ListItem component from Material UI does accepted the prop button..we should change it if we don't want to use the ButtonBase only : 
![image](https://user-images.githubusercontent.com/29228205/67859926-08192600-fb1d-11e9-87ce-5a028365d532.png)

So IMO it's fine if you leave like it is now...we will check all cases in our design system

Reply:
https://material-ui.com/api/list-item/

Reply:
regarding this `lint` rule, I believe we should discuss about it. I find this rule very annoying ... I would also prefer to write `button` only